### PR TITLE
tests: Add thread suspend-resume SMP stress test

### PR DESCRIPTION
**Pull request by peter-mitsis**
Adds a test to stress k_thread_suspend() and k_thread_resume() on an SMP system. The test should take about 30 seconds.

Issue #58392 highlighted flaws in k_thread_suspend()/k_thread_resume() on an SMP system. As those race conditions have since been fixed, it is worth while to have a test that stresses those routines.

### CODE REVIEW

**Review by dcpleung — APPROVED** [review]

File: tests/kernel/smp_suspend/src/main.c [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+		k_thread_create(&thread[i], thread_stack[i],
+				STACK_SIZE, thread_entry,
+				i < (NUM_THREADS - 1) ? &thread[i + 1] : NULL,
+				(void *)(uintptr_t)i, NULL,
+				10 - i, 0, K_FOREVER);
+
+		k_thread_suspend(&thread[i]);
+
+		k_thread_start(&thread[i]);
```
Comment (andyross) [inline tests/kernel/smp_suspend/src/main.c:61]: Might be worth a comment explaining here that you want the thread "launched" in a suspended state so it doesn't run until k_thread_resume() is called.

That said: that's a good edge case to test but it probably wouldn't hurt to be exercising an async k_thread_suspend() too, given that this is a stress test.

File: tests/kernel/smp_suspend/src/main.c [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+	/* Create the threads */
+
+	printk("Starting ...\n");
+
+	for (i = 0; i < NUM_THREADS; i++) {
+
+		k_thread_create(&thread[i], thread_stack[i],
+				STACK_SIZE, thread_entry,
+				i < (NUM_THREADS - 1) ? &thread[i + 1] : NULL,
```
Comment (andyross) [inline tests/kernel/smp_suspend/src/main.c:55]: Maybe "(i + 1) % NUM_THREADS"?  Or is there a reason you want the final thread to have a null resume_thread argument?

File: tests/kernel/smp_suspend/src/main.c [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+	/*
+	 * All newly created test threads are currently in the suspend state.
+	 * Start the first thread.
+	 */
+
+	k_thread_resume(&thread[0]);
+
+	for (unsigned int iteration = 0; iteration < 30; iteration++) {
+		k_sleep(K_MSEC(1000));
```
Comment (andyross) [inline tests/kernel/smp_suspend/src/main.c:72]: Seems like a needlessly long sleep?  The goal here is just to check that the counters keep incrementing, right?  Seems like faster iterations for less time might be a better choice?

**Review by andyross — APPROVED** [review]
Seems reasonable, notes & nitpicks.

File: tests/kernel/smp_suspend/src/main.c (hunk 3 above)
Comment (cfriedt) [reply tests/kernel/smp_suspend/src/main.c:72]: Would just yield work?

**Review by cfriedt — APPROVED** [review]

**Pull request merged**